Userland/Libraries/LibWeb/HTML/Scripting/Fetching.cpp

```diff
@@ -280,6 +280,50 @@ static void set_up_module_script_request(Fetch::Infrastructure::Request& request
     request.set_priority(options.fetch_priority);
 }
 
+// https://html.spec.whatwg.org/multipage/webappapis.html#get-the-descendant-script-fetch-options
+WebIDL::ExceptionOr<ScriptFetchOptions> get_descendant_script_fetch_options(ScriptFetchOptions original_options, const URL::URL& url, EnvironmentSettingsObject& settings_object)
+{
+    // 1. Let newOptions be a copy of originalOptions.
+    ScriptFetchOptions new_options;
+    new_options.cryptographic_nonce = original_options.cryptographic_nonce;
+    new_options.integrity_metadata = original_options.integrity_metadata;
+    new_options.parser_metadata = original_options.parser_metadata;
+    new_options.credentials_mode = original_options.credentials_mode;
+    new_options.referrer_policy = original_options.referrer_policy;
+    new_options.render_blocking = original_options.render_blocking;
+    new_options.fetch_priority = original_options.fetch_priority;
```

Comment:
Do we need to copy each member like this or would something like `auto new_options = original_options` work?